test/unit/Cardano/WalletLayerSpec.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.WalletLayerSpec
+    ( spec
+    ) where
+
+
+import Prelude
+
+import Cardano.DBLayer
+    ( DBLayer (..), PrimaryKey (..) )
+import Cardano.DBLayer.MVar
+    ( newDBLayer )
+import Cardano.NetworkLayer.HttpBridge
+    ( newNetworkLayer )
+import Cardano.Wallet
+    ( WalletId (..), WalletName (..) )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase (..) )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap, SeqState )
+import Cardano.Wallet.Mnemonic
+    ( Entropy
+    , EntropySize
+    , Mnemonic
+    , MnemonicException (..)
+    , MnemonicWords
+    , ambiguousNatVal
+    , entropyToMnemonic
+    , mkEntropy
+    )
+import Cardano.WalletLayer
+    ( NewWallet (..), WalletLayer (..), mkWalletLayer )
+import Control.Monad.IO.Class
+    ( liftIO )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Crypto.Encoding.BIP39
+    ( ValidChecksumSize, ValidEntropySize, ValidMnemonicSentence )
+import Data.Either
+    ( isLeft, isRight )
+import Data.Maybe
+    ( isJust )
+import Test.Hspec
+    ( Spec, describe, it, shouldSatisfy )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , InfiniteList (..)
+    , Property
+    , arbitraryBoundedEnum
+    , checkCoverage
+    , choose
+    , vectorOf
+    )
+import Test.QuickCheck.Monadic
+    ( monadicIO )
+
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.List as L
+import qualified Data.Text as T
+
+
+spec :: Spec
+spec = do
+    describe "WalletLayer works as expected" $ do
+        it "Wallet upon creation is written down in db"
+            (checkCoverage walletCreationProp)
```

Comment:
Does it makes sense to use `checkCoverage` when there is no `cover` or `coverTable` used in the property?

Reply:
@piotr-iohk Nope it doesn't :)

Reply:
Without any `cover` or `coverTable`, it's equivalent to doing just `property`.